### PR TITLE
Retrieve network and device details

### PR DESCRIPTION
**Pull request by Meatballs1**
There was no processing of the status or network info messages that are sent back from devices. These include details such as device serial number, software versions, mac address, ip addresses. 

The way the 'topics' were created for the state meant that many of them were initialized as `None` and are never populated. 

I've modified the approach to store state for any key we are sent back as long as it is not initially None. If it becomes None at a future point we will overwrite the state. It may be better just to store the Nones that come back initially as presumably they may be populated later?

I've also refactored the initialization with the device, so it isn't just happening magically within the `handle_message` loop, but explicitly within the `reconnect()` method.

I've also introduced some basic checking of versions for services from the service data that is sent, and we reject messages to services that are unknown on that device. Currently we just spray and pray requests to endpoints on devices and hope they come back with data rather than do anything intelligent about working out which ones we need to talk to. 

Tests
- [ ] We should see MQTT state updates including device serial numbers, IP Addresses etc if available
- [ ] We should see state updates from running devices with updates on the progress of a program

### CODE REVIEW

**Comment by Meatballs1** [comment]
Without the filtering we do get some values identified by numbers only. I've checked the  devices.json file but they dont appear there, and I cant see them in the client/server communications (I dont have a rooted device to check local comms). They may get sent down in a message I haven't captured - may need to check local storage or something.

**Comment by Meatballs1** [comment]
I believe this will also bring in status updates that currently appear to be missed. Running this properly over the weekend I get:

```
app_1  | 2024-03-23 23:50:59.007683 dishwasher {'StartInRelative': 7440}
app_1  | 2024-03-23 23:51:59.058021 dishwasher {'StartInRelative': 7380}
app_1  | 2024-03-23 23:52:59.013440 dishwasher {'StartInRelative': 7320}
app_1  | 2024-03-23 23:53:59.057841 dishwasher {'StartInRelative': 7260}
app_1  | 2024-03-23 23:54:59.017107 dishwasher {'StartInRelative': 7200}
app_1  | 2024-03-23 23:55:59.060720 dishwasher {'StartInRelative': 7140}
app_1  | 2024-03-23 23:56:59.017606 dishwasher {'StartInRelative': 7080}
app_1  | 2024-03-23 23:57:59.058254 dishwasher {'StartInRelative': 7020}
app_1  | 2024-03-23 23:58:59.021392 dishwasher {'StartInRelative': 6960}
app_1  | 2024-03-23 23:59:59.062878 dishwasher {'StartInRelative': 6900}
app_1  | 2024-03-24 00:00:59.016699 dishwasher {'StartInRelative': 6840}
app_1  | 2024-03-24 00:01:59.072735 dishwasher {'StartInRelative': 6780}
app_1  | 2024-03-24 00:02:59.021945 dishwasher {'StartInRelative': 6720}
app_1  | 2024-03-24 00:03:59.067971 dishwasher {'StartInRelative': 6660}
app_1  | 2024-03-24 00:04:59.019024 dishwasher {'StartInRelative': 6600}
app_1  | 2024-03-24 00:05:59.061575 dishwasher {'StartInRelative': 6540}
app_1  | 2024-03-24 00:06:59.063643 dishwasher {'StartInRelative': 6480}
app_1  | 2024-03-24 00:07:59.071545 dishwasher {'StartInRelative': 6420}
app_1  | 2024-03-24 00:08:59.019791 dishwasher {'StartInRelative': 6360}
app_1  | 2024-03-24 00:09:59.073262 dishwasher {'StartInRelative': 6300}
app_1  | 2024-03-24 00:10:59.020916 dishwasher {'StartInRelative': 6240}
app_1  | 2024-03-24 00:11:59.069193 dishwasher {'StartInRelative': 6180}
app_1  | 2024-03-24 00:12:59.025978 dishwasher {'StartInRelative': 6120}
app_1  | 2024-03-24 00:13:59.066557 dishwasher {'StartInRelative': 6060}
```

But as its not a topic that is in the preset values that are currently setup by https://github.com/hcpy2-0/hcpy/pull/36/files#diff-3f989a0da0dfdf21149587cceb530c66278de34355d1019e8334eed372e5dee2L150-L153 as the definition only contains:

```
            "558": {
                "name": "BSH.Common.Option.StartInRelative"
```

File: HCDevice.py [hunk 1]
```diff
@@ -252,6 +265,47 @@ def get(self, resource, version=1, action="GET", data=None):
...
+
+        # the clothes washer wants this, the token doesn't matter,
+        # although they do not handle padding characters
+        # they send a response, not sure how to interpet it
+        self.token = base64url_encode(get_random_bytes(32)).decode("UTF-8")
+        self.token = re.sub(r"=", "", self.token)
+        self.get("/ci/authentication", version=2, data={"nonce": self.token})
+
+        self.get("/ci/info")  # clothes washer
+        self.get("/iz/info")  # dish washer
```
Comment (Meatballs1) [inline HCDevice.py:292]: We should check that `iz` `ci` etc are in services. We shouldn't send them due to checks further in the code but clearer to check here too.

**Comment by Meatballs1** [comment]
@pmagyar will you have a chance to look over these? Have been running these last few pull requests for a while now and seems stable and to bring back most/all of the data needed for automation

**Comment by pmagyar** [comment]
hey sorry, I already checked the code but wanted to test it myself. But I didn't get around to it over the Easter holidays. If you say that a long-term test was successful for you, then we can merge it.

**Review by pmagyar — APPROVED** [review]

**Comment by Meatballs1** [comment]
It would be good to have it tested with some other devices first

**Comment by pmagyar** [comment]
i have a dishwasher. I planned to take the test this week.

**Comment by pmagyar** [comment]
think it worked for me too. good job.
Thanks for your engagement!

**Pull request merged**